runelite-client/src/main/java/net/runelite/client/plugins/itemidentification/ItemIdentificationPlugin.java

```diff
@@ -54,11 +70,49 @@ ItemIdentificationConfig getConfig(ConfigManager configManager)
 	protected void startUp()
 	{
 		overlayManager.add(overlay);
+		clientThread.invokeLater(() ->
+		{
+			switch (client.getGameState())
+			{
+				case STARTING:
+				case UNKNOWN:
+					return false;
+			}
+
+			//Placeholder values cannot be obtained in a static context
+			//and must therefore be added on plugin startup.
+			addPlaceholderIds();
+			return true;
+		});
 	}
 
 	@Override
 	protected void shutDown()
 	{
 		overlayManager.remove(overlay);
+		placeholderIds.clear();
+	}
+
+	private void addPlaceholderIds()
+	{
+		if (!placeholderIds.isEmpty())
+		{
+			return;
+		}
+
+		for (ItemIdentification item : ItemIdentification.values())
+		{
+			for (int id : item.itemIDs)
+			{
+				int placeholderId = itemManager.getItemComposition(id).getPlaceholderId();
+				placeholderIds.put(placeholderId, item);
+			}
+		}
+	}
+
+	ItemIdentification findItemIdentification(int itemID)
+	{
+		final ItemIdentification identification = ItemIdentification.get(itemID);
+		return identification != null ? identification : placeholderIds.getOrDefault(itemID, null);
```

Comment:
Actually, to simplify this I would just get itemComposition here directly and checking it's placeholderId instead of caching it on startup

Reply:
What would be the best way to do this? Since `getItemComposition` must be called on `clientThread` i cached the values in order to access them in a relatively efficient fashion. I don't understand how to not use the map while without rendering the entire overlay on `clientThread`, which we obviously don't want to (can't?) do. I'm not too familiar with threading.

Reply:
Overlay render logic is already on clientThread so that should be fine